stable/datadog/templates/deployment.yaml

```diff
@@ -1,25 +1,37 @@
+{{- if .Values.deployment.enabled }}
 {{- if .Values.datadog.apiKey }}
-{{- if .Values.datadog.collectEvents }}
 apiVersion: extensions/v1beta1
 kind: Deployment
 metadata:
-  name: {{ template "fullname" . }}-events
+  name: {{ template "fullname" . }}
   labels:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
 spec:
-  replicas: 1
+  replicas: {{ .Values.deployment.replicas }}
   template:
     metadata:
       labels:
-        app: {{ template "fullname" . }}-events
-      name: {{ template "fullname" . }}-events
+        app: {{ template "fullname" . }}
+      name: {{ template "fullname" . }}
     spec:
       containers:
-      - name: {{ .Chart.Name }}-events
+      - name: {{ .Chart.Name }}
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         resources:
 {{ toYaml .Values.resources | indent 12 }}
+        ports:
+        - containerPort: 8125
```

Comment:
Great question @seanknox !

The first reason is **money**. Each team is a separate organization with it's own Datadog account and API key. They are responsible for their budget, as well as for their custom, application-specific dashboards and alerts.

The second reason is the **flexibility**. Using per-deployment ConfigMap team provides it's own settings, it's own python checks, which are specific to an application run and maintained by a specific team. I mean checks run in "super" namespace make sense for "superApp" run by "superTeam", while they make no sense for "megaTeam" running "megaApp" in "mega" namespace, which is a totally different application, sending totally different metrics, maybe even operating different protocols (one team has XMPP, another - HTTP etc).

*While the global cluster team is a separate unit that just builds clusters and does not care much about application specifics, has its own budget and fine-grained (ha ha, I hope) responsibilities.*